src/Readm/ReadmParser.ts

```diff
@@ -0,0 +1,265 @@
+  });
+
+  const tagSections: TagSection[] = [createTagSection({ id: '0', label: 'genres', tags: arrayTags.map(x => createTag(x)) })];
+
+  let status = MangaStatus.ONGOING;
+  switch (rawStatus.toLocaleUpperCase()) {
+    case 'ONGOING':
+      status = MangaStatus.ONGOING;
+      break;
+    case 'COMPLETED':
+      status = MangaStatus.COMPLETED;
+      break;
+    default:
+      status = MangaStatus.ONGOING;
+      break;
+  }
+
+  return createManga({
+    id: mangaId,
+    titles: [title],
+    image,
+    rating: Number(rating),
+    status: status,
+    author: author,
+    artist: artist,
+    tags: tagSections,
+    desc: description!,
+    hentai: hentai,
+  });
+}
+
+export const parseChapters = ($: CheerioStatic, mangaId: string): Chapter[] => {
+  const chapters: Chapter[] = [];
+
+  for (const elem of $("div.season_start").toArray()) {
+    const title = $("h6.truncate", elem).first().text().trim() ?? "";
+    const chapterId = $('a', elem).attr('href')?.split(mangaId + "/").pop() ?? '';
+    const chapterNumber = title.split("Chapter ")[1].split("-")[0];
+    const date = parseDate($("td.episode-date", elem)?.text() ?? "");
+
+    chapters.push(createChapter({
+      id: chapterId,
+      mangaId,
+      name: title,
+      langCode: LanguageCode.ENGLISH,
+      chapNum: Number(chapterNumber),
+      time: date,
+    }))
+  }
+
+  return chapters;
+}
+
+export const parseChapterDetails = ($: CheerioStatic, mangaId: string, chapterId: string): ChapterDetails => {
+  const pages: string[] = [];
+
+  for (const p of $("div.ch-images img").toArray()) {
+    let rawPage = $(p).attr("src");
+    rawPage = "https://readm.org" + rawPage;
+    pages.push(rawPage);
+  }
+
+  let chapterDetails = createChapterDetails({
+    id: chapterId,
+    mangaId: mangaId,
+    pages: pages,
+    longStrip: false
+  });
+  return chapterDetails;
+}
+
+export const parseTags = ($: CheerioStatic): TagSection[] | null => {
+  let arrayTags: Tag[] = [];
+  for (const tag of $("li", "ul.trending-thisweek.categories").toArray()) {
+    const label = $("a", tag).text().trim();
+    const id = $("a", tag).attr('href')?.split("category/")[1] ?? "";
+    arrayTags.push({ id: id, label: label });
+  }
+  const tagSections: TagSection[] = [createTagSection({ id: '0', label: 'genres', tags: arrayTags.map(x => createTag(x)) })];
+  return tagSections;
+}
+
+export interface UpdatedManga {
+  ids: string[],
+  loadMore: boolean;
+}
+
+export const parseUpdatedManga = ($: CheerioStatic, time: Date, ids: string[]): UpdatedManga => {
+  const updatedManga: string[] = [];
+  let loadMore = true;
+
+  for (let p of $("div.poster.poster-xs", $("ul.clearfix.latest-updates").first()).toArray()) {
+    const id = $('a', p).attr('href')?.split('manga/').pop() ?? '';
+    const mangaTime = parseDate($("span.date", p).text().trim() ?? "");
+    if (mangaTime > time)
+      if (ids.includes(id))
+        updatedManga.push(id);
+      else loadMore = false;
```

Comment:
else is paired with the wrong if
```suggestion
    if (mangaTime > time) {
      if (ids.includes(id)) {
        updatedManga.push(id);
      }
    } else { 
      loadMore = false;
    }
```

Reply:
What he's got here 'technically' might work, but as a coding convention, the suggestion is much nicer.